src/hotspot/share/opto/vtransform.cpp

```diff
@@ -40,36 +40,73 @@ void VTransformGraph::add_vtnode(VTransformNode* vtnode) {
     }                                                 \
   )
 
-// This is similar to IGVN optimization. But we are a bit lazy, and don't care about
-// notification / worklist, since the list of nodes is rather small, and we don't
-// expect optimizations that trickle over the whole graph.
-void VTransformGraph::optimize(VTransform& vtransform) {
-  TRACE_OPTIMIZE( tty->print_cr("\nVTransformGraph::optimize"); )
-
-  bool progress = true;
-  DEBUG_ONLY(int pass_count = 0;)
-  while (progress) {
-    progress = false;
-    assert(++pass_count < 10, "ensure we do not have endless loops");
-    for (int i = 0; i < _vtnodes.length(); i++) {
-      VTransformNode* vtn = _vtnodes.at(i);
-      if (!vtn->is_alive()) { continue; }
-      progress |= vtn->optimize(_vloop_analyzer, vtransform);
-
-      // Nodes that have no use any more are dead.
-      if (vtn->out_strong_edges() == 0 &&
-          // There are some exceptions:
-          // 1. Memory phi uses are not modeled, so they appear to have no use here, but must be kept alive.
-          // 2. Similarly, some stores may not have their memory uses modeled, but need to be kept alive.
-          // 3. Outer node with strong inputs: is a use after the loop that we must keep alive.
-          !(vtn->isa_PhiScalar() != nullptr ||
-            vtn->is_load_or_store_in_loop() ||
-            (vtn->isa_Outer() != nullptr && vtn->has_strong_in_edge()))) {
-        vtn->mark_dead();
-        progress = true;
-      }
-    }
+void VTransformOptimize::worklist_push(VTransformNode* vtn) {
+  if (_worklist_set.test_set(vtn->_idx)) { return; }
+  _worklist.push(vtn);
```

Comment:
I would flip this since it's only one line:

```suggestion
  if (!_worklist_set.test_set(vtn->_idx)) {
    _worklist.push(vtn);
  }
```